api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_online.py

```diff
@@ -74,12 +92,21 @@ def __init__(self, n_samples: int, n_variables: int):
 
         self.u = np.zeros(n_variables, dtype=np.float64)
         self.n_samples = n_samples
+        # If we want to exclude zeros, we need to keep track of the denominator
+        self.nnz_only = nnz_only
+        self.n = np.zeros(n_variables)
 
     def update(self, var_vec: npt.NDArray[np.int64], val_vec: npt.NDArray[np.float32]) -> None:
-        _update_mean_vector(self.u, var_vec, val_vec)
+        if self.nnz_only:
+            _update_mean_and_denominator_vectors(self.u, self.n, var_vec, val_vec)
```

Comment:
I'm still finding the use of `denominator` inconsistent, but I'll stop bugging you about it :-)

Reply:
I can rename it to `_update_mean_and_n_vectors`.